src/nbl/video/CSurfaceVKWin32.cpp

```diff
@@ -0,0 +1,18 @@
+#include "nbl/video/surface/CSurfaceVKWin32.h"
+
+#include "nbl/video/CVulkanConnection.h"
+
+namespace nbl {
+namespace video
+{
+
+core::smart_refctd_ptr<CSurfaceVKWin32> CSurfaceVKWin32::create(IAPIConnection* api, SCreationParams&& params)
+{
+    auto* vk = static_cast<CVulkanConnection*>(api);
```

Comment:
you should assert or check that the `IAPIConnection`'s type matches the expected (pretty much on every object now, you need to check)